### ISSUE TITLE
Extreme Artifacting in Grimblast Screenshot

### ISSUE THREAD

**Issue by TZProgrammer** [hyprwm/Hyprland#4029]
### Hyprland Version

Hyprland, built from branch main at commit 80b9b21f9f24b6e8db2fc6f7705cd124f436ffba dirty (opengl: fix nvidia read formats). Tag: v0.32.3-105-g80b9b21f  flags: (if any) debug

### Bug or Regression?

Bug

### Description

When using Grimblast to take a screenshot, extreme artifacting happens. This might or might not be an Nvidia issue.

### How to reproduce

For me, I only had to take a screenshot.

![image](https://github.com/hyprwm/Hyprland/assets/22439185/a4c4f01e-c383-4ca5-a714-6ccb4bdc0d44)

**Comment by FlyingButteryTuna** [comment]
Yep, same here. Both on wayshot and grim

**Comment by iynaix** [comment]
Same problem, but on top of that, the screenshot on my vertical monitor is also squished and blank on the bottom half.

![image](https://github.com/hyprwm/Hyprland/assets/94313/eb35b53f-7e17-4164-aa35-e664a9401459)

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/hyprwm/Hyprland/files/13539129/patch.txt)
check this

**Comment by TZProgrammer** [comment]
I have applied the patch but no luck. Just to make sure, as it is my first time doing this, do I just move the patch.txt to the Hyprland directory, run "patch -p1 > patch.txt", then "make all && sudo make install", and then try screenshot? I did get a message when applying the patch "File subprojects/wlroots is not a regular file -- refusing to patch" using this approach though.

**Comment by vaxerski** [comment]
`patch -p1 < patch.txt`...

**Comment by vaxerski** [comment]
also can someone bisect this...? Or try the patch? (unless yall dont want this fixed, obv)

**Comment by FlyingButteryTuna** [comment]
Can confirm 8440a30231ea41a6b1dacc60a28837b265a6efec is the last commit that doesn't have that issue
Also the path seems to fix the issue on my end

**Comment by FlyingButteryTuna** [comment]
Yep, works as expected now

**Comment by vaxerski** [comment]
what about latest...?

**Comment by vaxerski** [comment]
waaait, are those glitches nvidia only?

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/hyprwm/Hyprland/files/13540695/patch.txt)
?

**Comment by FlyingButteryTuna** [comment]
Nope, that just brings back the color issues with bitdepth 8 lol:
![1701649446-wayshot](https://github.com/hyprwm/Hyprland/assets/67516242/8d2ae13c-6a90-4848-98fd-55a5f96478dd)
vs what it actually looks like
![image](https://github.com/hyprwm/Hyprland/assets/67516242/f4823a6b-71d5-4821-95fd-30251bbec111)

**Comment by FlyingButteryTuna** [comment]
bitrate 8 is fine here, but with with bitrate 10 I can't even take a screenshot:
![image](https://github.com/hyprwm/Hyprland/assets/67516242/03838a9a-72f2-41c9-8b0b-0d7686074a3e)

**Comment by vaxerski** [comment]
hm. Means nvidia doesn't support 10bit formats that grim supports and vice versa. What we could do is just force nvidia to screencopy at 8bit...

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/hyprwm/Hyprland/files/13540940/patch.txt)
?

**Comment by FlyingButteryTuna** [comment]
That did not work, the artifacts are still there, but the colors are just a bit different now.

I played around with the changes introduced in 9c09f2a847a1372c73cf67dc9e7e9ebe309b142e and then found out that if I rebase to dc2082b00ac0a477df32dd75a32a2a0d8ad71c90 and introduce back this chunk of code in screencopy (line 459) and toplevelexport line (389)
```
const auto PFORMAT = gles2FromDRM(format);
...
glReadPixels(0, 0, frame->box.width, frame->box.height, PFORMAT->gl_format, PFORMAT->gl_type, data);
```
then artifacts (and blank spaces too) don't happen anymore
8bit:
![1701655704-wayshot](https://github.com/hyprwm/Hyprland/assets/67516242/1f52912a-96fa-4ed2-a645-3a91169ffc75)

10bit:
![1701655717-wayshot](https://github.com/hyprwm/Hyprland/assets/67516242/b62f6c88-aaca-4316-82bf-685a51b86062)
*sigh*

**Comment by vaxerski** [comment]
[patch.txt](https://github.com/hyprwm/Hyprland/files/13541579/patch.txt)
try
